### PR TITLE
Adding verbose mode and support for coveralls from Jenkins

### PR DESCRIPTION
**Pull request by laurentheirendt**
This PR combines PR https://github.com/JuliaCI/Coverage.jl/pull/164 and the request from @ararslan to implement a verbose mode for coveralls.

Similar to PR https://github.com/JuliaCI/Coverage.jl/pull/165, I added the support for coveralls from Jenkins.

cc: @syarra

### CODE REVIEW

File: src/coveralls.jl [hunk 1]
```diff
@@ -11,7 +11,6 @@ module Coveralls
     using HTTP
     using JSON
     using Compat
-    using Compat.LibGit2
```
Comment (ararslan) [inline src/coveralls.jl]: Why was this removed? I'm fairly certain this is still needed.

Comment (laurentheirendt) [reply src/coveralls.jl]: Is `LibGit2` not part of `Base`? (https://docs.julialang.org/en/stable/devdocs/libgit2/index.html)

Comment (ararslan) [reply src/coveralls.jl]: It is in Julia 0.6 but not in 0.7. In 0.6, `using Compat.LibGit2` is the same as `using Base.LibGit2`, and in 0.7 it's the same as `using LibGit2`, which is part of the standard library rather than Base. Either way, this is still required.

Comment (laurentheirendt) [reply src/coveralls.jl]: I see ... Thanks for clarifying! I will add it back in ...

**Comment by laurentheirendt** [comment]
@ararslan, may you merge this PR? Thanks 👍
